backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CAdapterGenerator.kt

```diff
@@ -382,6 +383,17 @@ internal class CAdapterGenerator(
     private val scopes = mutableListOf<ExportedElementScope>()
     internal val prefix = context.config.moduleId
     private lateinit var outputStreamWriter: PrintWriter
+    private val paramNamesRecorded = mutableMapOf<String, Int>()
+
+    internal fun paramsToUniqueNames(params: List<ParameterDescriptor>): List<String> {
+        paramNamesRecorded.clear()
+        return params.map {
+            val name = translateName(it.name.asString()) 
+            val count = paramNamesRecorded.getOrDefault(name, 0)
+            paramNamesRecorded[name] = count + 1
+            "$name${count.toString().takeUnless { it == "0" } ?: ""}"
```

Comment:
Looks like it is somewhat optimized for the uncommon case of name collision.

Reply:
Yes that code is somewhat golfed. I'll fix.